test/integration/integration.test.ts

```diff
@@ -109,493 +65,260 @@ describe('Integration tests', (): void => {
-    const receiverWalletAddressUrl =
-      'https://happy-life-bank-test-backend:4100/accounts/pfry'
-    const amountValueToSend = '500'
-
-    let gfranklinWalletAddressId: string
-    let receiver: ReceiverGql
-    let quote: QuoteGql
-    let outgoingPayment: OutgoingPaymentGql
-
-    beforeAll(async () => {
-      const gfranklinWalletAddress = await c9.accounts.getByWalletAddressUrl(
+    test('Peer to Peer - Cross Currency', async (): Promise<void> => {
+      const {
+        createReceiver,
+        createQuote,
+        createOutgoingPayment,
+        getOutgoingPayment
+      } = testActions.admin
+
+      const senderWalletAddress = await c9.accounts.getByWalletAddressUrl(
         'https://cloud-nine-wallet-test-backend:3100/accounts/gfranklin'
       )
-      assert(gfranklinWalletAddress?.walletAddressID)
-      gfranklinWalletAddressId = gfranklinWalletAddress.walletAddressID
-    })
-
-    test('Create Receiver (remote Incoming Payment)', async (): Promise<void> => {
-      const handleWebhookEventSpy = jest.spyOn(
-        hlb.integrationServer.webhookEventHandler,
-        'handleWebhookEvent'
-      )
-      const response = await c9.adminClient.createReceiver({
+      assert(senderWalletAddress)
+      const senderAssetCode = senderWalletAddress.assetCode
+      const senderWalletAddressId = senderWalletAddress.walletAddressID
+      const value = '500'
+      const createReceiverInput = {
         metadata: {
-          description: 'For lunch!'
+          description: 'cross-currency'
         },
         incomingAmount: {
-          assetCode: 'USD',
+          assetCode: 'EUR',
           assetScale: 2,
-          value: amountValueToSend as unknown as bigint
+          value: value as unknown as bigint
         },
-        walletAddressUrl: receiverWalletAddressUrl
+        walletAddressUrl:
+          'https://happy-life-bank-test-backend:4100/accounts/lars'
+      }
+
+      const receiver = await createReceiver(createReceiverInput)
+      assert(receiver.incomingAmount)
+
+      const quote = await createQuote(senderWalletAddressId, receiver)
+      const outgoingPayment = await createOutgoingPayment(
+        senderWalletAddressId,
+        quote
+      )
+      const payment = await getOutgoingPayment(outgoingPayment.id, value)
+
+      const receiverAssetCode = receiver.incomingAmount.assetCode
+      const exchangeRate =
+        hlb.config.seed.rates[senderAssetCode][receiverAssetCode]
+      const fee = c9.config.seed.fees.find((fee: Fee) => fee.asset === 'USD')
+
+      // Expected amounts depend on the configuration of asset codes, scale, exchange rate, and fees.
+      assert(receiverAssetCode === 'EUR')
+      assert(senderAssetCode === 'USD')
+      assert(
+        receiver.incomingAmount.assetScale === senderWalletAddress.assetScale
+      )
+      assert(senderWalletAddress.assetScale === 2)
+      assert(exchangeRate === 0.91)
+      assert(fee.fixed === 100)
```

Comment:
do we need to `assert(fee)` here beforehand? 
I'm getting `'fee' is possibly 'undefined'` here

Reply:
fixed [52b9f19](https://github.com/interledger/rafiki/pull/2595/commits/52b9f19e443f5186e778caf0c021ed2d9a885f1c)